.github/workflows/protobuf-dockerimage.yml

```diff
@@ -0,0 +1,35 @@
+name: Protobuf Docker Image
+on: 
+  push:
+    tags: [ '**' ]
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+    paths: 
+      - .github/workflows/protobuf-dockerimage.yml
+      - protobuf/Dockerfile
+      - protobuf/protoc-wrapper
+          
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build the Docker image
+      run: docker build protobuf/. -t build-protobuf
+    - name: Push the Docker image
+      if: github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags/')
+      run: |
+        echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+        function tag_and_push {
+          docker tag build-protobuf "otel/build-protobuf:${1}" && docker push "otel/build-protobuf:${1}"
+        }
+        if [[ "${GITHUB_REF}" == "refs/heads/master" ]]; then
+          tag_and_push "latest"
+        elif [[ "${GITHUB_REF}" =~ refs/tags/v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+            TAG="${GITHUB_REF#"refs/tags/v"}"
+            tag_and_push "${TAG}"
+            tag_and_push "latest"
```

Comment:
what if it's a tag for an old release (like a hotfix)? Should it still push for latest?

Reply:
Done.